File: app/src/main/java/com/walhalla/data/repository/FavoritePresenter.java
package com.walhalla.data.repository;

import android.content.Context;
import android.os.Handler;

import com.walhalla.data.model.Channel;

import java.util.List;

public class FavoritePresenter extends BasePresenter {

    public FavoritePresenter(Handler handler, Context context) {
        super(handler, context);
    }

    public void getFavorite(long playlistId, RepoCallback<List<Channel>> callback) {
        executeInBackground(() -> {
            try {
                List<Channel> channels = db_repo.getFavorite(playlistId);
                postToMainThread(() -> callback.successResult(channels));
            } catch (Exception e) {
                postToMainThread(() -> callback.errorResult(e.getMessage()));
            }
        });
    }
}

File: app/src/main/java/com/walhalla/data/repository/FavoritePresenter.kt
package com.walhalla.data.repository

import android.content.Context
import android.os.Handler
import com.walhalla.data.model.Channel

class FavoritePresenter(handler: Handler, context: Context) : BasePresenter(handler, context) {
    fun getFavorite(playlistId: Long, callback: RepoCallback<List<Channel>>) {
        executeInBackground(Runnable {
            try {
                val channels: List<Channel> = db_repo.getFavorite(playlistId)
                postToMainThread { callback.successResult(channels) }
            } catch (e: Exception) {
                postToMainThread { callback.errorResult(e.message!!) }
            }
        })
    }
}
